File: easy/primepal.js
function isPrime(n) {
  if (n < 2) {
    return false;
  } else if (n === 2) {
    return true;
  } else {
    for (var i = 2; i <= Math.sqrt(n); i++) {
      if (n % i === 0) {
        return false;
      }
    }
  }
  return true;
}

function isPalindrome(str) {
  if (str.length <= 1) {
    return true;
  } else {
    if (str[0] === str.slice(-1)) {
      isPalindrome(str.slice(1,-1));
      return true;
    } else {
      return false;
    }
  }
}

function primePal() {
  var best = 0;
  for (var n = 1; n <= 1000; n++) {
    if (isPrime(n) && isPalindrome(n.toString())) {
      best = n;
    }
  }
  console.log(best);
}

primePal();

File: easy/primepal.rb
def is_prime?(n)
  return false if n < 2
  return true if n == 2
  (2..Math.sqrt(n)).each { |num| return false if n % num == 0 }
  true
end

def prime_pal
  best = 0
  (1..1000).each do |n|
    if is_prime?(n) && n.to_s == n.to_s.reverse
      best = n
    end
  end
  puts best
end

prime_pal